File: agents/screeps-agent/workspace/kimi/place_spawn.js
// Script to place spawn in available room
const http = require('http');

const rooms = ['W1N1', 'W1N3', 'W3N1', 'W3N3', 'W5N1', 'W5N3', 'W5N5'];
const positions = [
    {x: 10, y: 10}, {x: 15, y: 15}, {x: 20, y: 20}, 
    {x: 25, y: 25}, {x: 30, y: 30}, {x: 35, y: 35}
];

const token = '825a47dc6595309b1be4598c11d6b276863b1590';

function tryPlaceSpawn(room, x, y) {
    return new Promise((resolve, reject) => {
        const postData = JSON.stringify({
            room: room,
            x: x,
            y: y,
            name: "Spawn1"
        });

        const options = {
            hostname: 'localhost',
            port: 21025,
            path: '/api/game/place-spawn',
            method: 'POST',
            headers: {
                'Content-Type': 'application/json',
                'Content-Length': Buffer.byteLength(postData),
                'X-Token': token,
                'X-Username': 'kimi'
            }
        };

        const req = http.request(options, (res) => {
            let data = '';
            res.on('data', (chunk) => {
                data += chunk;
            });
            
            res.on('end', () => {
                const response = JSON.parse(data);
                console.log(`Trying ${room} at ${x},${y}: ${JSON.stringify(response)}`);
                
                if (response.ok === 1) {
                    console.log(`🎉 SUCCESS! Spawn placed in ${room} at ${x},${y}`);
                    resolve(true);
                } else {
                    resolve(false);
                }
            });
        });

        req.on('error', (e) => {
            console.error(`Error trying ${room} at ${x},${y}: ${e.message}`);
            resolve(false);
        });

        req.write(postData);
        req.end();
    });
}

async function findAvailableSpawn() {
    for (let room of rooms) {
        for (let pos of positions) {
            const success = await tryPlaceSpawn(room, pos.x, pos.y);
            if (success) {
                console.log(`Spawn successfully placed!`);
                return;
            }
            // Small delay between attempts
            await new Promise(resolve => setTimeout(resolve, 100));
        }
    }
    console.log('No available spawn locations found');
}

findAvailableSpawn();

File: agents/screeps-agent/workspace/gemini/place_spawn.py

import urllib.request
import json
import time
import find_spot

def get_token():
    url = "http://localhost:21025/api/auth/signin"
    data = json.dumps({"email":"gemini@test.com","password":"password"}).encode('utf-8')
    req = urllib.request.Request(url, data=data, headers={'Content-Type': 'application/json'})
    with urllib.request.urlopen(req) as response:
        resp = json.loads(response.read().decode('utf-8'))
        return resp.get('token')

def place_spawn(token, room, x, y):
    url = "http://localhost:21025/api/game/place-spawn"
    data = json.dumps({"room": room, "x": x, "y": y, "name": "Spawn1"}).encode('utf-8')
    headers = {
        'X-Token': token,
        'X-Username': 'gemini',
        'Content-Type': 'application/json'
    }
    req = urllib.request.Request(url, data=data, headers=headers)
    try:
        with urllib.request.urlopen(req) as response:
            return json.loads(response.read().decode('utf-8'))
    except urllib.error.HTTPError as e:
        return {"error": str(e)}

def main():
    token = get_token()
    print(f"Token: {token}")
    
    room = "W5N5"
    spots = find_spot.find_spots()
    
    if not spots:
        print("No spots found!")
        return

    print(f"Found {len(spots)} spots. Trying the best ones...")
    
    for x, y in spots:
        print(f"Trying {x}, {y}...")
        res = place_spawn(token, room, x, y)
        if res.get('ok') == 1:
            print(f"SUCCESS at {x}, {y}!")
            return
        else:
            print(f"Failed: {res}")
            if "invalid room" in str(res):
                 # maybe wait a bit?
                 pass
        time.sleep(0.1)

if __name__ == "__main__":
    main()
